Mage/src/main/java/mage/abilities/ActivatedAbilityImpl.java

```diff
@@ -204,7 +206,13 @@ public ActivationStatus canActivate(UUID playerId, Game game) {
         //  game.inCheckPlayableState() can't be a help here cause some cards checking activating status,
         //  activatorId must be removed
         this.activatorId = playerId;
-        return new ActivationStatus(true, approvingObject);
+
+        if (approvingObjects.isEmpty()) {
+            return ActivationStatus.withoutApprovingObject(true);
+        }
+        else {
+            return new ActivationStatus(approvingObjects);
```

Comment:
Sorry, one more question- why no `true` parameter here anymore?

Reply:
Won't be able to look at the actual code for a day or two, but ActivationStatus really has three modes:

- Some approvers, it is the role of the caller if any of them is actually good enough (mana to activate, further restriction, etc...)
- Not approved (actually two way to construct that for convenience, the asFalse constructor, and the withoutApprover set with the false parameter)
- Approved, but without an explicit approver. I did not want to mess much with those cases as I did not understand all of it and was not the main focus point of the PR (being fixing a few bugs, and allow Court of Lochtwain to work without workaround).

Reply:
Sure, that's fine, I just noticed that the old line of code had the true parameter and the new code doesn't (in the else case with approving objects not empty), so I wanted to confirm that was intentional.

Reply:
It is intentional, and to be completely clear, the same behavior is preserved, but split in two cases:
If there is no approvingObject, the ActivationStatus is still forced to be approved.
If there is at least one approvingObject, the Approving Status being non-empty will expose that it is approved without the need to have a true parameter to force the status to approved.